src/main/scala/com/wavesplatform/state2/StateStorage.scala

```diff
@@ -17,9 +17,14 @@ class StateStorage private(file: Option[File]) extends VariablesStorage(createMV
 
   override protected val Version = 2
 
+  @volatile private[state2] var lastUpdated: Long = 0L
+
   def getHeight: Int = getInt(heightKey).getOrElse(0)
 
-  def setHeight(i: Int): Unit = putInt(heightKey, i)
+  def setHeight(i: Int): Unit = {
+    putInt(heightKey, i)
+    lastUpdated = System.currentTimeMillis
```

Comment:
Why not use `Time`? It is not critical here, but it's best if we use single source of time everywhere